File: src/main/java/com/brenoepic/clothes/Clothing.java
package com.brenoepic.clothes;

import lombok.Getter;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.Objects;

@Getter
public class Clothing
{
    private final int id;
    private final String type;
    private final String set;
    private final int effect;
    public Clothing(final ResultSet result) throws SQLException {
        this.id = result.getInt("id");
        this.type = result.getString("type");
        if(Objects.equals(this.type, "figure")) this.set = result.getString("figure");
        else this.set = result.getString("setid");
        this.effect = result.getInt("effect");
    }
}

File: src/main/java/io/github/brenoepics/clothes/Clothing.kt
package io.github.brenoepics.clothes

import java.sql.ResultSet

class Clothing
    (result: ResultSet) {
    val id: Int = result.getInt("id")
    val type: String = result.getString("type")
    var set: String? = null
    val effect: Int

    init {
        if (this.type == "figure") this.set = result.getString("figure")
        else this.set = result.getString("setid")
        this.effect = result.getInt("effect")
    }
}
